Rename forall, exists and find predicate and operator params.

janekdb opened the pull request:
Align parameters names to use p for predicates and op for combining operations.

Based on #4760. Extended to include,
- Tuple2Zipped
- Tuple3Zipped
- Either

The original author was vsalvis.

The scope for this commit is code changes with some closely related documentation moved over.

janekdb commented:
Review by @sschaef, @lrytz, @SethTisue, @som-snytt, @soc.

kiritsuku commented on src/library/scala/collection/GenTraversableOnce.scala:
```diff
@@ -393,20 +393,35 @@ trait GenTraversableOnce[+A] extends Any {
+  def forall(@deprecatedName('pred) p: A => Boolean): Boolean
 
-  def exists(pred: A => Boolean): Boolean
+  /** Tests whether a predicate holds for some of the elements of this $coll.
```
s/for some/at least for one/

SethTisue replied:
"for at least one" is more idiomatic

som-snytt replied:
"for one, at least" is more plaintive.

I think Scala needs a wiki page of existential puns and expressions.

janekdb replied:
Fixed here and in `TraversableLike`

kiritsuku commented on src/library/scala/collection/LinearSeqOptimized.scala:
```diff
@@ -117,20 +117,20 @@ trait LinearSeqOptimized[+A, +Repr <: LinearSeqOptimized[A, Repr]] extends Linea
 
   override /*TraversableLike*/
-  def foldLeft[B](z: B)(f: (B, A) => B): B = {
+  def foldLeft[B](z: B)(@deprecatedName('f) op: (B, A) => B): B = {
```
Why `op`? Is this commonly used in other libraries too?

mpociecha replied:
+1 (I agree with Simon that f is okay in this case)

SethTisue replied:
most of the other places `foldLeft` is declared in the collections, the name `op` is used. so Janek is just bringing a few outliers in line.

som-snytt replied:
+1 for `op`. I wish I knew why op feels more natural. Is it because we're assuming a binop?

janekdb replied:
Also aligned in a private object in `RedBlackTree.scala`. The access to this object is `private` so voided the need for `deprecatedName`.

som-snytt commented on src/library/scala/collection/GenTraversableOnce.scala:
```diff
@@ -393,20 +393,35 @@ trait GenTraversableOnce[+A] extends Any {
+   *  @param p       the predicate used to test elements.
    *  @return        an option value containing the first element in the $coll
-   *                 that satisfies `p`, or `None` if none exists.
+   *                 for which `p` returns `true`, or `None` if none exists.
```
FTR, satisfying a predicate is much more math-nerdy than having something return true. Personally, I prefer the original.

More importantly, you really can't say "none exists." You can say, "if none satisfies p", etc. "otherwise None" would probably suffice.

Ichoran replied:
I also prefer the "satisfies" `p` terminology.  It is shorter and more precise, and can be figured out without undue hardship.

janekdb replied:
Reverted.

som-snytt commented on src/library/scala/collection/immutable/Set.scala:
```diff
@@ -71,7 +71,7 @@ object Set extends ImmutableSetFactory[Set] {
     def - (elem: Any): Set[Any] = this
     def iterator: Iterator[Any] = Iterator.empty
-    override def foreach[U](f: Any =>  U): Unit = {}
+    override def foreach[U](f: Any => U): Unit = {}
```
Great whitespace change, but I'd also change the RHS to `()`, which is most expressive and idiomatic. So long as you're touching a LOC.

janekdb replied:
RHS changed to `()`.

FTR occurrences that remain

| Style | Count |
| --- | --: |
| Unit = {} | 35 |
| Unit = () | 202 |

janekdb committed:
Rename forall, exists and find predicate and operator params.

Align parameters names to use p for predicates and op for combining operations.

Based on #4760. Extended to include,

 - Tuple2Zipped
 - Tuple3Zipped
 - Either

The original author was vsalvis.

janekdb commented on src/library/scala/collection/immutable/RedBlackTree.scala:
```diff
@@ -409,11 +409,11 @@ object RedBlackTree {
     def cons[B](x: B, xs: NList[B]): NList[B] = new NList(x, xs)
 
-    def foldLeft[A, B](xs: NList[A], z: B)(f: (B, A) => B): B = {
+    def foldLeft[A, B](xs: NList[A], z: B)(op: (B, A) => B): B = {
```
This object this method belongs to has `private[this]` access so I'm assuming `deprecatedName` is not required.

janekdb commented:
Finished responding to comments.

Review by @sschaef, @SethTisue, @som-snytt, @mpociecha .

SethTisue commented:
LGTM

janekdb commented:
Once this is merged I can rebase #4760 and continue with the other changes.
